src/myinfo-types.ts

```diff
@@ -1,160 +1,391 @@
 export enum MyInfoSource {
-    GovtVerified = '1',
-    UserProvided = '2',
-    NotApplicable = '3',
-    SingPassVerified = '4',
+  GovtVerified = '1',
+  UserProvided = '2',
+  NotApplicable = '3',
+  SingPassVerified = '4',
 }
 
-export enum MyInfoSex {
-    Male = 'M',
-    Female = 'F',
-    Unknown = 'U',
+interface IMetadata {
+  lastupdated: string
+  source: MyInfoSource
+  classification: 'C'
+  unavailable?: boolean
 }
 
-interface IMyInfoMetadata {
-    lastUpdated: string
-    source: MyInfoSource
-    classification: 'C'
+interface OptionalString {
+  value?: string
 }
 
-export interface IMyInfoBasicField extends IMyInfoMetadata {
-    value?: string
+interface OptionalNumber {
+  value?: number | ''
 }
 
-export interface IMyInfoHighLowField extends IMyInfoMetadata {
-    high: number
-    low: number
+interface OptionalBoolean {
+  value?: boolean | ''
 }
```

Comment:
we should re-look at the types here as well. it doesn't look correct, but it would reflect quite badly on the MyInfo API if they returned empty strings each time a value was missing.

Reply:
it's right at least according to the MockPass sample personas. for example, if you look at `EXPECTED_PERSON_DATA` in `test/constants.ts`, `merdekagen.eligibility.value` should be a `boolean`, but an empty string is returned.